android/java/org/chromium/chrome/browser/crypto_wallet/util/ImageLoader.java

```diff
@@ -6,46 +6,284 @@
+                        Rect bestSize = new Rect(0, 0, 0, 0);
+                        while (iterBitmap.hasNext() && iterSize.hasNext()) {
+                            Bitmap bitmap = iterBitmap.next();
+                            Rect size = iterSize.next();
+                            if (size.width() > bestSize.width()
+                                    && size.height() > bestSize.height()) {
+                                bestBitmap = bitmap;
+                                bestSize = size;
+                            }
+                        }
+                        if (bestSize.width() == 0 || bestSize.height() == 0) {
+                            imageFetcherFacade = null;
+                        } else {
+                            BitmapDrawable bitmapDrawable = new BitmapDrawable(bestBitmap);
+                            imageFetcherFacade = new ImageFetcherFacade(bitmapDrawable);
+                        }
+                        loadImage(imageFetcherFacade, context, isCircular, imageView, callback);
+                    });
+        } else {
+            ImageFetcher imageFetcher = ImageFetcherFactory.createImageFetcher(
+                    ImageFetcherConfig.NETWORK_ONLY, profile.getProfileKey());
+            if (isGif(url)) {
+                imageFetcher.fetchGif(Params.create(new GURL(url), "unused"), gifImage -> {
+                    ImageFetcherFacade imageFetcherFacade =
+                            new ImageFetcherFacade(gifImage.getData());
+                    loadImage(imageFetcherFacade, context, isCircular, imageView, callback);
+                });
+            } else {
+                imageFetcher.fetchImage(Params.create(new GURL(url), "unused"), bitmap -> {
+                    BitmapDrawable bitmapDrawable = new BitmapDrawable(bitmap);
+                    ImageFetcherFacade imageFetcherFacade = new ImageFetcherFacade(bitmapDrawable);
+                    loadImage(imageFetcherFacade, context, isCircular, imageView, callback);
+                });
+            }
+        }
+    }
+
+    /**
+     * Parses an input string as an XML document, manipulates it to add the missing attributes to
+     * the svg tag, and then serializes it back to a string. If the input string is not a valid SVG
+     * image, the method returns null.
+     * <p>
+     * Note that this implementation assumes that the input string is a valid SVG image and does
+     * not perform any further validation or sanitation of the contents of the image.
+     *
+     * @param input Input string to sanitize.
+     * @return sanitized string with 'width' and 'height' attributes set in the 'svg' tag.
+     */
+    public static String sanitizeSvg(String input) {
+        try {
+            // Convert the input string to an XML document
+            DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+            DocumentBuilder builder = factory.newDocumentBuilder();
```

Comment:
Disabled `DOCTYPE` declarations. Addressed in commit [`0858190` (#17330)](https://github.com/brave/brave-core/pull/17330/commits/08581900aa4d465d37ac571d0af58d29896de4d8)

Reply:
We should test this to not call any external service whenever we hit a local picture. It's typical for XML parsers to call external services, whenever some special input (DOCTYPE, XXE, external resources)

Reply:
We should check exactly what `FEATURE_SECURE_PROCESSING` does and doesn't fix for `DocumentBuilder`

Reply:
@stoletheminerals , I'm afraid I can't confirm as I've never extensively used this class. Do you want me to check further? Would `XmlPullParser` work better for some reasons?

Reply:
@simoarpe `DocumentBuilderFactory` with `FEATURE_SECURE_PROCESSING` may suffice. But xmlpullparser is better in terms of avoiding XXE and other XML related attacks (such as https://en.wikipedia.org/wiki/Billion_laughs_attack) in general since it doesn't parse entities or DTD. I'd suggest to explore if we can use it instead, just for extra security.

Reply:
It turned out `XmlPullParser` is not very flexible for writing back values (that is our use case). 
So here is the code I put together, let me know if it may work, and I'll finish tweaking the last bits.
It's not very elegant, and it relies on the `XmlSerializer` for writing back values.

```java
    public static String sanitizeSvg(String input) {
        try {
            // Create an XmlPullParser instance
            XmlPullParserFactory factory = XmlPullParserFactory.newInstance();
            XmlPullParser parser = factory.newPullParser();

            // Set input source for parser
            parser.setInput(new StringReader(input));

            // Find start tag
            int eventType = parser.getEventType();
            while (eventType != XmlPullParser.START_TAG) {
                eventType = parser.next();
            }

            // Check if the start tag is an SVG tag
            if (!parser.getName().equalsIgnoreCase("svg")) {
                // Input is not an SVG image.
                return null;
            }

            // Check if the SVG tag has width and height attributes
            boolean widthFound = false;
            boolean heightFound = false;
            String width = null;
            String height = null;

            for (int i = 0; i < parser.getAttributeCount(); i++) {
                String attrName = parser.getAttributeName(i);
                String attrValue = parser.getAttributeValue(i);

                if (attrName.equalsIgnoreCase("width")) {
                    widthFound = true;
                    width = attrValue;
                } else if (attrName.equalsIgnoreCase("height")) {
                    heightFound = true;
                    height = attrValue;
                }
            }

            if (widthFound && heightFound) {
                // Width and height attributes are declared.
                // There's no need to modify the current SVG.
                return input;
            }

            // Add missing width and height attributes to SVG tag
            if (!widthFound) {
                width = "1000px";
            }

            if (!heightFound) {
                height = "1000px";
            }

            // Serialize the modified XML document to a string and return it
            XmlSerializer serializer = XmlPullParserFactory.newInstance().newSerializer();
            StringWriter writer = new StringWriter();
            serializer.setOutput(writer);

            while (eventType != XmlPullParser.END_DOCUMENT) {
                if (eventType == XmlPullParser.START_TAG) {
                    serializer.startTag(null, parser.getName());

                    for (int i = 0; i < parser.getAttributeCount(); i++) {
                        String name = parser.getAttributeName(i);
                        String value = parser.getAttributeValue(i);
                        serializer.attribute(null, name, value);
                    }
                    if (parser.getName().equalsIgnoreCase("svg")) {
                        if (!widthFound) {
                            serializer.attribute(null, "width", width);
                        }
                        if (!heightFound) {
                            serializer.attribute(null, "height", height);
                        }
                    }
                } else if (eventType == XmlPullParser.TEXT) {
                    serializer.text(parser.getText());
                } else if (eventType == XmlPullParser.END_TAG) {
                    serializer.endTag(null, parser.getName());
                } else if (eventType == XmlPullParser.IGNORABLE_WHITESPACE) {
                    serializer.ignorableWhitespace(parser.getText());
                }

                eventType = parser.next();
            }

            serializer.flush();
            return writer.toString();
        } catch (XmlPullParserException | IOException e) {
            return null;
        }
    }
```

Reply:
@simoarpe looks good to me. LMK when it's implemented so I can dynamically test it to double check.

Reply:
That code doesn't do anything to sanitise the following:

```xml
<?xml version="1.0" standalone="yes"?>
<!DOCTYPE test [ <!ENTITY xxe SYSTEM "file:///etc/hosts" > ]>
<svg width="128px" height="128px" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1">
  <text font-size="16" x="0" y="16">&xxe;</text>
</svg>
```

@stoletheminerals do you have some SVG test cases you are planning to test with?

Reply:
Oh sorry @simoarpe  I see you are still tweaking it

Reply:
> That code doesn't do anything to sanitise the following:
> 
> 
> 
> ```xml
> 
> <?xml version="1.0" standalone="yes"?>
> 
> <!DOCTYPE test [ <!ENTITY xxe SYSTEM "file:///etc/hosts" > ]>
> 
> <svg width="128px" height="128px" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1">
> 
>   <text font-size="16" x="0" y="16">&xxe;</text>
> 
> </svg>
> 
> ```
> 
> 
> 
> @stoletheminerals do you have some SVG test cases you are planning to test with?

Nothing in particular, just normal XXE payloads as the one you provided. Did you have a chance to test it with xmlpullparser? It shouldn't require sanitizing I believe, as dtd and entity processing is disabled by default.